### DISCUSSION TITLE
Sonoff mini + smart light bulb

### DISCUSSION THREAD

**Discussion by leandromagro** [RavenSystem/esp-homekit-devices#1803] [Q&A]
Hello everyone.
I would like to have some smart bulbs at home but i dont want to remove all sonoff mini installed inside the wall or change my dumb switches.
What do you think about changing the behavior of the sonoff mini from swtich to stateless button? Do you think this is going to work?
Any other idea for this scenario? Actual: dumb switch + sonoff mini (as a smart switch) + smart bulb.
Thanks in advance.

**Comment by andyallcorn** [comment]
I guess an important question is: what are you hoping to achieve? Just to disable the Sonoff so that the smart bulb is always on? That’s easy enough; you don’t even have to have a switch appear in homekit, you could have a set of json that hid it using h:0, set the initial state to be on, and remove the input button controls so it could never be turned off.

Or, were you hoping to have the switch control the bulb on/off (without cutting power)? That would be tricky, but if your smart bulb of choice had a REST API it _might_ be possible by changing the Sonoff to have network actions instead of setting GPIOs. Would depend which bulbs you chose. (FYI, it should be possible in theory with an HAA-compatible bulb - the insecure rest api could be enabled for the bulb - but that isn’t recommended in the wiki.)

**Reply by leandromagro** [reply]
Hi thanks for the answer.
Just trying to figure out in a brainstorming.

The ideia is to keep the physical switch in the wall as the sonoff inside it controlling the smart bulb without cutting the electricity on the out put port. And for that, i thought to use a set of rules in the eve app: if the light is on and physical switch changes the state of the sonoff's S1 or S2 ports (with or without electricity), then turn it off. The same for the opposite way. Always via software.

**Reply by andyallcorn** [reply]
Hmm, yeah, that could work — I’ve had mixed success using HomeKit triggers, so am nervous of them… but I think a lot of that has been because of the particularly niche things I’ve been trying to achieve! Hence me thinking network actions to skip the HK triggers.

Going the HK way, I thiiiink I’d split the switch into two switches, switch (A) with the button input but no GPIO actions; switch (B) with the GPIO actions but no controls. Then set up your triggers in Eve on the HomeKit-visible switch (A). Meanwhile, switch (B) would have an initial state of on, and would be hidden from HK with "h":0 — leaving the power on and no way to turn it off. There may be a more elegant way (eg. action on boot to set the GPIO, removing the need for (B)) but that’s the easiest conversion that occurs to me.
